x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextField.java

```diff
@@ -296,9 +297,23 @@ public static List<Chunk> toSemanticTextFieldChunks(
                 for (var chunk : textExpansionResults.getChunkedResults()) {
                     chunks.add(new Chunk(chunk.matchedText(), toBytesReference(contentType.xContent(), chunk.weightedTokens())));
                 }
-            } else if (result instanceof ChunkedTextEmbeddingResults textEmbeddingResults) {
-                for (var chunk : textEmbeddingResults.getChunks()) {
-                    chunks.add(new Chunk(chunk.matchedText(), toBytesReference(contentType.xContent(), chunk.embedding())));
+            } else if (result instanceof ChunkedTextEmbeddingFloatResults textEmbeddingFloatResults) {
```

Comment:
Every time a new chunked text embedding class is added, the client code needs to take it into account.

Could all these result classes inherit from the same class / interface, included the chunks, so we don't have to make changes but rely on polymorphism?

If every chunked result has a `getChunks()` method that returns a common `Chunk` base class or interface, and is able to serialize, we could avoid having these ugly `instanceof` cases and the need to modify `semantic_text` code to account for new classes.

Reply:
That makes sense to me. I've tried to implement this solution, please take a look when you get the chance.